src/uct/ib/base/ib_md.c

```diff
@@ -1610,10 +1620,19 @@ static void uct_ib_check_gpudirect_driver(uct_ib_md_t *md, const char *file,
 
 ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
                                    struct ibv_device *ib_device,
-                                   const uct_ib_md_config_t *md_config)
+                                   const uct_ib_md_config_t *md_config,
+                                   size_t memh_base_size, size_t mr_size)
 {
     ucs_status_t status;
 
+    status = uct_ib_md_parse_device_config(md, md_config);
+    if (status != UCS_OK) {
+        goto err;
+    }
+
+    uct_ib_md_parse_relaxed_order(md, md_config);
+    uct_ib_md_init_memh_size(md, memh_base_size, mr_size);
```

Comment:
usually we do function-call initialization after setting the fields (after line 1647)

Reply:
done